app/models/movie_credits.rb

```diff
@@ -11,32 +13,29 @@ def initialize(adult, credit_id, department, job, tmdb_id, title, poster_path, d
     @poster_path = poster_path
     @date = date
     @character = character
+  end
 
-  end #init
-
-  attr_accessor :adult, :credit_id, :department, :job, :tmdb_id, :title, :poster_path, :date, :character
-
-  def self.parse(json)
-    @credits = []
-    json.each do |result|
-      @adult = result[:adult]
-      @credit_id = result[:credit_id]
-      @department = result[:department]
-      @job = result[:job]
-      @tmdb_id = result[:id]
-      @title = result[:title]
-      @poster_path = result[:poster_path]
-      if result[:release_date].present?
-        @date = Date.parse(result[:release_date]).stamp("2001")
-      else
-        @date = "Date unavailable"
-      end
-      @character = result[:character] if result[:character].present?
+  class << self
+    def parse(results)
+      results.map do |result|
+        new(
+          adult: result[:adult],
+          credit_id: result[:credit_id],
+          department: result[:department],
+          job: result[:job],
+          tmdb_id: result[:id],
+          title: result[:title],
+          poster_path: result[:poster_path],
+          date: format_release_date(result[:release_date]),
+          character: result[:character]
+        )
+      end.sort_by { |credit| credit.date }.reverse
+    end
 
-      @credit = MovieCredits.new(@adult, @credit_id, @department, @job, @tmdb_id, @title, @poster_path, @date, @character)
-      @credits << @credit
-    end #loop
-    @credits = @credits.sort_by{ |credit| credit.date }.reverse
-  end #parse
+    private
 
-end #class
\ No newline at end of file
+    def format_release_date(date_data)
```

Comment:
Since this method is private, I moved it into the `private` section to help us understand this object's interface.